lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,648 @@
+    pending_messages_limit: usize,
+}
+
+impl ChunkedGelfDecoder {
+    /// Creates a new `ChunkedGelfDecoder`.
+    pub fn new(timeout_millis: u64, pending_messages_limit: usize) -> Self {
+        Self {
+            bytes_decoder: BytesDecoder::new(),
+            state: Arc::new(Mutex::new(HashMap::new())),
+            timeout: Duration::from_millis(timeout_millis),
+            pending_messages_limit,
+        }
+    }
+
+    /// Decode a GELF chunk
+    pub fn decode_chunk(&mut self, mut chunk: Bytes) -> Result<Option<Bytes>, BoxedFramingError> {
+        // Encoding scheme:
+        //
+        // +------------+-----------------+--------------+----------------------+
+        // | Message id | Sequence number | Total chunks |    Chunk payload     |
+        // +------------+-----------------+--------------+----------------------+
+        // | 64 bits    | 8 bits          | 8 bits       | remaining bits       |
+        // +------------+-----------------+--------------+----------------------+
+        //
+        // As this codec is oriented for UDP, the chunks (datagrams) are not guaranteed to be received in order,
+        // nor to be received at all. So, we have to store the chunks in a buffer (state field) until we receive
+        // all the chunks of a message. When we receive all the chunks of a message, we can concatenate them
+        // and return the complete payload.
+
+        // We need 10 bytes to read the message id, sequence number and total chunks
+        if chunk.remaining() < 10 {
+            let src_display = format!("{chunk:?}");
+            warn!(message = "Received malformed chunk headers (message ID, sequence number and total chunks) with less than 10 bytes. Ignoring it.",
+                src = src_display,
+                remaining = chunk.remaining(),
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+        let message_id = chunk.get_u64();
+        let sequence_number = chunk.get_u8();
+        let total_chunks = chunk.get_u8();
+
+        if total_chunks == 0 || total_chunks > GELF_MAX_TOTAL_CHUNKS {
+            warn!(
+                message = "Received a chunk with an invalid total chunks value. Ignoring it.",
+                message_id = message_id,
+                sequence_number = sequence_number,
+                total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        if sequence_number >= total_chunks {
+            warn!(
+                message = "Received a chunk with a sequence number greater than total chunks. Ignoring it.",
+                message_id = message_id,
+                sequence_number = sequence_number,
+                total_chunks = total_chunks,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        let mut state_lock = self.state.lock().unwrap();
```

Comment:
```suggestion
        let mut state_lock = self.state.lock().expect("poisoned lock");
```

Nit: we prefer to use `expect` rather than `unwrap`.